backend/src/apis/fetch.rs

```diff
@@ -1,151 +1,102 @@
+    let sum_eth_balance = get_balances
+        .into_iter()
+        .reduce(|acc, balance| acc + balance)
+        .unwrap();
+    result.push(sum_eth_balance);
+
+    // Most in case, the number of contracts is less than the number of asset addresses
+    // Iterating over contracts first is more efficient
+    let mut sum_erc_balance = U256::zero();
+    for contract in &contracts {
+        for addr in asset_addresses.clone() {
+            let erc_balance = contract
+                .get_balance_from_contract(addr)
+                .call()
+                .await
+                .unwrap();
+            sum_erc_balance = sum_erc_balance + erc_balance;
         }
+        result.push(sum_erc_balance)
     }
-    Ok(balances)
+
+    Ok(result)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[test]
-    #[ignore]
-    fn test_fetch_balances_per_addr() {
-        // this is an address with 0 usdc and 0.010910762665574143 ETH
-        let address = "0xe4D9621321e77B499392801d08Ed68Ec5175f204".to_string();
+    use crate::contracts::tests::initialize_anvil;
 
-        let usdc_contract = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48".to_string();
-        let eth_contract = "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee".to_string();
+    use ethers::utils::Anvil;
 
-        let balances =
-            fetch_balances_per_addr(address, [usdc_contract, eth_contract].to_vec()).unwrap();
+    #[tokio::test]
+    async fn test_fetch_asset_sums() {
+        // Necessary to implement `get_balance_from_contract` for the `contracts` parameter by following trait
+        impl<M: Middleware> BalanceFromContract<M> for MockERC20<M> {
+            fn get_balance_from_contract(&self, account: Address) -> ContractCall<M, U256> {
+                self.balance_of(account)
+            }
+        }
 
-        assert_eq!(balances[0], 0); // balance for usdc
-        assert_eq!(balances[1], 10910762665574143); // balance for wei
-    }
+        let anvil: ethers::utils::AnvilInstance = Anvil::new()
+            .mnemonic("test test test test test test test test test test test junk")
+            .spawn();
 
-    #[test]
-    #[ignore]
-    fn test_fetch_asset_sums() {
-        let address_1 = "0xe4D9621321e77B499392801d08Ed68Ec5175f204".to_string(); // this is an address with 0 usdc and 0.010910762665574143 ETH
-        let address_2 = "0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1".to_string(); // this is an address with 0.000001 USDC usdc and 1 wei
+        let (cex_addr_1, cex_addr_2, client, mock_erc20) = initialize_anvil(&anvil).await;
 
-        let usdc_contract = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48".to_string();
-        let eth_contract = "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee".to_string();
+        // send RPC requests with `anvil_setBalance` method via provider to adjust balance of `cex_addr_1` and `cex_addr_2`
+        for addr in [cex_addr_1, cex_addr_2].to_vec() {
+            let _res = client
```

Comment:
I think setting the ETH balances can also go into the `initialize_anvil` function, since you're already initializing the ERC20 balances there.

Reply:
Moved to `initialize_anvil` at https://github.com/summa-dev/summa-solvency/commit/9ada335025190df0da7e21ebfc361ee2ab68fdd5